files/acl/acl.php

```diff
@@ -1,11 +1,59 @@
 <?php
 
+/**
+ * Plugin Name: VIP Restricted Files
+ * Description: Secure your content by restricting access to unpublished or private files.
+ * Author: WordPress VIP
+ * Author URI: https://wpvip.com
+ * Version: 1.0
+ */
+
 namespace Automattic\VIP\Files\Acl;
 
 const FILE_IS_PUBLIC = 'FILE_IS_PUBLIC';
 const FILE_IS_PRIVATE_AND_ALLOWED = 'FILE_IS_PRIVATE_AND_ALLOWED';
 const FILE_IS_PRIVATE_AND_DENIED = 'FILE_IS_PRIVATE_AND_DENIED';
 
+add_action( 'muplugins_loaded', __NAMESPACE__ . '\maybe_load_restrictions' );
+
+function maybe_load_restrictions() {
+	$is_files_acl_enabled = defined( 'VIP_FILES_ACL_ENABLED' ) && VIP_FILES_ACL_ENABLED;
+	$is_restrict_all_enabled = get_option_as_bool( 'vip_files_acl_restrict_all_enabled' );
+	$is_restrict_unpublished_enabled = get_option_as_bool( 'vip_files_acl_restrict_unpublished_enabled' );
+
+	if ( ! $is_files_acl_enabled ) {
+		// Throw warning if restrictions are enabled but ACL constant is not set.
+		// This is probably a sign that options were copied between sites or someone missed a setup step.
+		if ( $is_restrict_all_enabled || $is_restrict_unpublished_enabled ) {
+			trigger_error( 'File ACL restrictions are enabled without server configs (missing `VIP_FILES_ACL_ENABLED` constant).', E_USER_WARNING );
+		}
+
+		return;
+	}
+
+	if ( $is_restrict_all_enabled ) {
+		require_once( __DIR__ . '/files/acl/restrict-all-files.php' );
+		return;
+	}
+
+	if ( $is_restrict_unpublished_enabled ) {
+		require_once( __DIR__ . '/files/acl/restrict-unpublished-files.php' );
+		return;
+	}
+}
+
+function get_option_as_bool( $option_name, $default = false ) {
+	$value = get_option_as_bool( 'vip_files_acl_restrict_all_enabled', false );
```

Comment:
@mjangda This reads like a bad copy-pasta?

Reply:
Yep :) Good catch!

Reply:
Probably need a test for this.

In the future, might be worth abstracting this out to a shared lib because I copied this from VIP Search.

Reply:
Done in 98616d4 and 6859d2a